### PR TITLE
Allow to connect specific plugs during installation

### PR DESCRIPTION
**Pull request by sergio-costas**
Sometimes it is needed to be able to connect some plugs during installation. Although the right way of doing it is by editing the gadget-desktop snap, this is cumbersome for quick tests.

This patch adds a quick&dirty way of connecting the plugs during installation. It basically creates a systemd service that is run at startup and tries to connect the specified plugs. When it succeeds, it automatically disables itself.

This method must be used only during tests, never for production.

### CODE REVIEW

**Review by robert-ancell — APPROVED** [review]
I've reviewed the code and while a little complex it looks good.

File: Makefile [hunk 1]
```diff
@@ -33,6 +33,7 @@ install:
 	/bin/cp -a static/* $(DESTDIR)
 	mkdir -p $(DESTDIR)/install-data
 	/bin/cp -r $(CRAFT_STAGE)/local-debs $(DESTDIR)/install-data/local-debs
+	$(CRAFT_PROJECT_DIR)/generate-connections.py $(CRAFT_PROJECT_DIR)/snap-connections.txt $(DESTDIR)/snap-connections.sh
```
Comment (robert-ancell) [inline Makefile:36]: Could this not be created directly into /usr/libexec?

Comment (sergio-costas) [reply Makefile:36]: Good point...

Comment (sergio-costas) [reply Makefile:36]: I really left it there because it's just "for testing purposes", but you are right: it's better to keep it clean.

Comment (sergio-costas) [reply Makefile]: Done.

**Review by robert-ancell — APPROVED** [review]
Easier to understand, thanks!

**Pull request merged**